protocols/relay/src/v2.rs

```diff
@@ -42,48 +42,53 @@ pub mod client {
 
         #[deprecated(
             since = "0.15.0",
-            note = "Use libp2p_relay::client::Transport instead."
+            note = "ClientTransport has been renamed to Transport
+            and is now private as it shouldn't be required by end users."
```

Comment:
I think they still need to interact with it right? We are returning it from `libp2p::relay::client::new`. Users should be allowed to name that one if they want.

We _could_ return an `impl Transport` actually but that is a bit more constraining. Esp. when users compose those transports together, they would lose the ability to name the type unless they use `.boxed`. In reality though, that is probably what they should do. Naming the entire chain of `.or`'d transports is super clunky.

Returning `impl Transport` would be efficient (no boxing) and would hide the type. I think that might be a good idea but I'd like to hear @mxinden's opinion on it!

Reply:
Changing to `impl Trait` now would be a breaking change so let's not do it and stick to doing just the rename!

Reply:
yeah right, I had forgotten that, thanks Thomas! Thou we probably can use the opaque type described on other cases.

Reply:
Interesting idea on using `impl Transport`. Rename is fine by me (for now).